Documentation/git-multi-pack-index.txt

```diff
@@ -39,6 +39,18 @@ expire::
 	Rewrite the MIDX file afterward to remove all references to
 	these pack-files.
 
+repack::
+	When given as the verb, collect a batch of pack-files whose
+	size are all at most the size given by --batch-size, but
+	whose sizes sum to larger than --batch-size. The batch is
```

Comment:
Will this starve?  That is, if I say --batch-size=10K and I only have 3 1K files, does it do anything?  Or does it wail until I have ~10 files?

Alternatively, if I have 30 1K files, does it try to combine all of them into 1?   Or just the first ~10 and let me run it again to get the next batch?

Reply:
> Will this starve? That is, if I say --batch-size=10K and I only have 3 1K files, does it do anything? Or does it wait until I have ~10 files?

It will wait until we have 10K worth of files.



> Alternatively, if I have 30 1K files, does it try to combine all of them into 1? Or just the first ~10 and let me run it again to get the next batch?

It grabs 10 and creates one new pack. A future run can collect the others.

Reply:
ok thanks